### PR TITLE
M2-4915 unit tests for scores extractor penetrating

### PR DESCRIPTION
**Pull request by anq83**
### 📝 Description

🔗 [Jira Ticket M2-4915](https://mindlogger.atlassian.net/browse/M2-4915)

This PR adds some new unit-tests that test ScoresExtractor module.

This PR changes:
-  adds some new unit-tests

### CODE REVIEW

File: src/features/pass-survey/model/tests/ScoresExtractor.test.ts [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+        checkBoxesItem.payload.options[2],
+      ];
+
+      const sliderAnswer: SliderResponse = 7;
+
+      const answers: Answers = {
+        0: { answer: radioAnswer },
+        1: {
+          answer: checkBoxAnswer,
+        },
```
Comment (farmerpaul) [inline src/features/pass-survey/model/tests/ScoresExtractor.test.ts:192]:
```suggestion
        1: { answer: checkBoxAnswer },
```

File: src/features/pass-survey/model/tests/ScoresExtractor.test.ts [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+        2: { answer: sliderAnswer },
+      };
```
Comment (farmerpaul) [inline src/features/pass-survey/model/tests/ScoresExtractor.test.ts:194]: I'm not sure I see the benefit of having this section (lines 167-194) copy-pasted for every test (it's the exact same chunk of code used 4 times); couldn't `pipelineItems` and `answers` also just be defined as static top-level constants, reused by all the tests?

Reduced redundancy would make the code easier to read and maintain.

**Review by farmerpaul — COMMENTED** [review]
Hi @anq83, tests look good to me; I just had a few comments mostly about reducing redundancy/improving code readability.

File: src/features/pass-survey/model/tests/ScoresExtractor.test.ts [hunk 3]
```diff
@@ -0,0 +1,471 @@
...
+  RadioResponse,
+  ScoreRecord,
+  SliderPipelineItem,
+  SliderResponse,
+} from '../../lib';
+import { ScoreConditionsEvaluator } from '../ScoreConditionsEvaluator';
+import { ScoresCalculator } from '../ScoresCalculator';
+import { ScoresExtractor } from '../ScoresExtractor';
+
+const getRadiosItem = (): RadioPipelineItem => {
```
[the remark this answers is not in the document]
Comment (anq83) [reply src/features/pass-survey/model/tests/ScoresExtractor.test.ts:19]: To isolate an object in scope of a test. New test ("it" block) = new object. What if logic of the tested module will change the test object?

**Review by BamMironov — APPROVED** [review]

**Pull request merged**